anchor/network/src/discovery.rs

```diff
@@ -0,0 +1,460 @@
+    /// always false.
+    pub started: bool,
+}
+
+impl Discovery {
+    pub async fn new(local_keypair: Keypair, network_config: &Config) -> Result<Self, String> {
+        let _enr_dir = match network_config.network_dir.to_str() {
+            Some(path) => String::from(path),
+            None => String::from(""),
+        };
+
+        // TODO info!(log, "ENR Initialised"; "enr" => local_enr.to_base64(), "seq" => local_enr.seq(), "id"=> %local_enr.node_id(),
+        //       "ip4" => ?local_enr.ip4(), "udp4"=> ?local_enr.udp4(), "tcp4" => ?local_enr.tcp4(), "tcp6" => ?local_enr.tcp6(), "udp6" => ?local_enr.udp6(),
+        //       "quic4" => ?local_enr.quic4(), "quic6" => ?local_enr.quic6()
+        // );
+
+        let discv5_listen_config =
+            discv5::ListenConfig::from_ip(Ipv4Addr::UNSPECIFIED.into(), 9000);
+
+        // discv5 configuration
+        let discv5_config = discv5::ConfigBuilder::new(discv5_listen_config).build();
+
+        // convert the keypair into an ENR key
+        let enr_key: CombinedKey = CombinedKey::from_libp2p(local_keypair)?;
+
+        let enr = build_enr(&enr_key, network_config).unwrap();
+        let mut discv5 = Discv5::new(enr, enr_key, discv5_config)
+            .map_err(|e| format!("Discv5 service failed. Error: {:?}", e))?;
+
+        // Add bootnodes to routing table
+        for bootnode_enr in network_config.boot_nodes_enr.clone() {
+            // TODO if bootnode_enr.node_id() == local_node_id {
+            //     // If we are a boot node, ignore adding it to the routing table
+            //     continue;
+            // }
+            // TODO debug!(
+            //     log,
+            //     "Adding node to routing table";
+            //     "node_id" => %bootnode_enr.node_id(),
+            //     "peer_id" => %bootnode_enr.peer_id(),
+            //     "ip" => ?bootnode_enr.ip4(),
+            //     "udp" => ?bootnode_enr.udp4(),
+            //     "tcp" => ?bootnode_enr.tcp4(),
+            //     "quic" => ?bootnode_enr.quic4()
+            // );
+
+            //let repr = bootnode_enr.to_string();
+            let _ = discv5.add_enr(bootnode_enr).map_err(|_e| {
+                // TODO  error!(
+                //     log,
+                //     "Could not add peer to the local routing table";
+                //     "addr" => repr,
+                //     "error" => e.to_string(),
+                // )
+            });
+        }
+
+        // Start the discv5 service and obtain an event stream
+        let event_stream = if !network_config.disable_discovery {
+            discv5.start().map_err(|e| e.to_string()).await?;
+            // TODO debug!(log, "Discovery service started");
+            EventStream::Awaiting(Box::pin(discv5.event_stream()))
+        } else {
+            EventStream::InActive
+        };
+
+        if !network_config.boot_nodes_multiaddr.is_empty() {
```

Comment:
Bootnodes MA might not be necessary. See https://github.com/ssvlabs/ssv-spec/blob/main/p2p/SPEC.md#discovery